File: net.akehurst.language.editor.demo/application/agl-editor-worker/src/jsMain/kotlin/worker.kt
import net.akehurst.language.agl.processor.Agl
import net.akehurst.language.editor.worker.AglSharedWorker


fun main() {
    val worker = AglSharedWorker<Any,Any>().also {
        Agl.registry.register(
            identity = "user-language",
            grammarStr = null,
            buildForDefaultGoal = false,
            aglOptions = Agl.options {
                semanticAnalysis {
                    active(false)
                }
            },
            configuration = Agl.configurationDefault()
        ).identity
    }
    worker.start()
}

File: net.akehurst.language.editor/browser/agl-editor-browser-agl/src/jsMain/resources/worker.js
importScripts("./require.min.js")
var w;
requirejs(['./net.akehurst.language.editor-agl-editor-browser-worker.js'], function(alg_worker) {
    const AglSharedWorker = alg_worker.net.akehurst.language.editor.worker.AglSharedWorker;
    w = new AglSharedWorker()
})